### PR TITLE
Fix for amplitude_bell size offset

### LINKED ISSUE #49: Amplitude anomaly fails with `amplitude_bell` size offset
Hi there 👋

first of all, thanks for the great repository. 
I came across this example which seems to fail:

```yaml
timeseries:
  - name: test-amp
    length: 1000
    base-oscillations:
      - kind: sine
    anomalies:
      - position: end
        length: 47
        channel: 0
        kinds:
          - kind: amplitude
            amplitude_factor: 2.0
```
Running with: `python -m gutenTAG --config-yaml tests/configs/config-amp.yaml --seed 42 --no-save --plot`
this fails to create the anomaly

```
Generating datasets:   0%|                                                                                      | 0/1 [00:00<?, ?it/s]
Traceback (most recent call last):
  File ".../.venv/lib/python3.11/site-packages/gutenTAG/gutenTAG.py", line 132, in generate
    results: List[Tuple[Dict, Dict[str, Any], Optional[List[ExtTimeSeries]]]] = Parallel(n_jobs=n_jobs)(
                                                                                ^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../.venv/lib/python3.11/site-packages/joblib/parallel.py", line 1863, in __call__
    return output if self.return_generator else list(output)
                                                ^^^^^^^^^^^^
  File ".../.venv/lib/python3.11/site-packages/joblib/parallel.py", line 1792, in _get_sequential_output
    res = func(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^
  File ".../.venv/lib/python3.11/site-packages/gutenTAG/gutenTAG.py", line 157, in internal_generate
    ts.generate(ctx.seed)
  File ".../.venv/lib/python3.11/site-packages/gutenTAG/generator/timeseries.py", line 37, in generate
    self.timeseries, self.labels = consolidator.generate(GenerationContext(seed=self._create_new_seed(random_seed)))
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../.venv/lib/python3.11/site-packages/gutenTAG/consolidator.py", line 43, in generate
    self.generate_anomalies(ctx)
  File ".../.venv/lib/python3.11/site-packages/gutenTAG/consolidator.py", line 67, in generate_anomalies
    anomaly_protocol = anomaly.generate(ctx.to_anomaly(current_base_oscillation, positions))
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../.venv/lib/python3.11/site-packages/gutenTAG/anomalies/__init__.py", line 65, in generate
    protocol = anomaly.generate(protocol)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../.venv/lib/python3.11/site-packages/gutenTAG/anomalies/types/amplitude.py", line 55, in generate
    subsequence = anomaly_protocol.base_oscillation.timeseries[anomaly_protocol.start:anomaly_protocol.end] * amplitude_bell
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
ValueError: operands could not be broadcast together with shapes (47,) (46,) 
```

Maybe I'm wrong but as it looks the error corresponds to
As far as I can see the error originates in:

https://github.com/TimeEval/GutenTAG/blob/b6e9e87c02c418f0603e9e91828f4fa5217e738a/gutenTAG/anomalies/types/amplitude.py#L35 and with `creeping-length` passed in
https://github.com/TimeEval/GutenTAG/blob/b6e9e87c02c418f0603e9e91828f4fa5217e738a/gutenTAG/anomalies/types/amplitude.py#L51

I'll open a PR soon, if this is fine for you.
Thanks again for the repo 👍

### PR DESCRIPTION
**Pull request by arrrrrmin**
As promised in https://github.com/TimeEval/GutenTAG/issues/49 here is the fix for the amplitude_bell size offset (fixes #49).
I’m not very familiar with unittest TestCase-Classes, @CodeLionX could you please review the `tests/test_generator/test_lengths.py`? 
There might be a more efficient way to check for the valid generation of anomalies under different length values?

Greetings and thanks for the very fast reply.

### CODE REVIEW

File: tests/test_generator/test_lengths.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                    ],
+                }
+            ]
+        }
+        self.creeping_factor = 0.3
+
+    def update_config(self, length: int, include_creeping: bool = False):
+        self.config["timeseries"][0]["anomalies"][0]["length"] = length
+        cl = int(round(length * self.creeping_factor)) if include_creeping else 0
+        self.config["timeseries"][0]["anomalies"][0]["creeping-length"] = cl
```
Comment (CodeLionX) [inline tests/test_generator/test_lengths.py:31]: I'm in favor of testing many lengths and abstracting this away as you do. However, I would avoid mutating members of the TestCase. I am unsure how this behaves when the tests are executed in parallel. To be on the safe side, please create a copy of `self.config`, mutate the copy, and just return the copy. You can then pass it to the `__run` function (see below).

File: tests/test_generator/test_lengths.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                            "channel": 0,
+                            "kinds": [{"kind": "amplitude", "amplitude_factor": 2.0}],
+                        }
+                    ],
+                }
+            ]
+        }
+        self.creeping_factor = 0.3
+
+    def update_config(self, length: int, include_creeping: bool = False):
```
Comment (CodeLionX) [inline tests/test_generator/test_lengths.py:28]: Please indicate that this method is private with `_`-prefix:
```suggestion
    def _update_config(self, length: int, include_creeping: bool = False) -> Dict[str, Any]:
```

File: tests/test_generator/test_lengths.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        self.config["timeseries"][0]["anomalies"][0]["length"] = length
+        cl = int(round(length * self.creeping_factor)) if include_creeping else 0
+        self.config["timeseries"][0]["anomalies"][0]["creeping-length"] = cl
+
+    def __run(self):
```
Comment (CodeLionX) [inline tests/test_generator/test_lengths.py:33]: We use single `_` to mark private methods.

You can use assertions in functions called from the original test function. I would have all the assertions in this function.

```suggestion
    def _run_and_validate(self, config: Dict[str, Any]) -> None:
```

File: tests/test_generator/test_lengths.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        gutentag = GutenTAG(seed=42)
+        gutentag.load_config_dict(self.config)
+        try:
+            gutentag.generate(return_timeseries=True)
+            res = True
+        except ValueError:
+            # For details see: https://github.com/TimeEval/GutenTAG/issues/49
+            # Catches ValueError:
+            #  'operands could not be broadcast together with shapes (n,) (n-1,)'
+            ...
```
Comment (CodeLionX) [inline tests/test_generator/test_lengths.py:44]: You can avoid catching the error in test cases. An error will automatically trigger a test failure. We can directly assert here if GutenTAG successfully generated the time series. Then, there is no need to track the results.
```suggestion
        ts = gutentag.generate(return_timeseries=True)
        self.assertIsNotNone(ts)
        self.assertEqual(len(ts), 1)
```

File: tests/test_generator/test_lengths.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        return res
+
+    def test_lengths(self):
+        results = []
+        for i in range(25, 50):
+            self.update_config(i, include_creeping=False)
+            results.append(self.__run())
+        assert all(results)
```
Comment (CodeLionX) [inline tests/test_generator/test_lengths.py:52]:
```suggestion
            config = self._update_config(i, include_creeping=False)
            self._run_and_validate(config))
```

Please change the other test accordingly.

**Review by CodeLionX — CHANGES_REQUESTED** [review]
Thank you very much for the prompt fix (+ test cases). I like the test but have some improvements regarding

File: tests/test_generator/test_lengths.py (hunk 1 above)
Comment (arrrrrmin) [reply tests/test_generator/test_lengths.py:31]: Thanks for the opportunity to learn about unittest with your comments 👍

**Comment by CodeLionX** [comment]
Mypy still fails on the test-file. You can add `# type: ignore` to the corresponding lines because specifying the full correct type of the config object is too verbose.

**Review by CodeLionX — APPROVED** [review]

**Pull request merged**